sklearn/neighbors/setup.py

```diff
@@ -20,6 +20,12 @@ def configuration(parent_package='', top_path=None):
                          include_dirs=[numpy.get_include()],
                          libraries=libraries)
 
+    config.add_extension('_nth_element',
```

Comment:
`_nth_element` is not the most descriptive name for those not familiar with c++. Maybe `_partition_nodes` would be better?